### PR TITLE
refactor(android): permission API cleanup changes

### PR DESCRIPTION
**Pull request by carlpoole**
Changes related to permission API cleanup https://github.com/ionic-team/capacitor/pull/4171

### CODE REVIEW

File: network/android/src/main/java/com/capacitorjs/plugins/network/NetworkPlugin.java [hunk 1]
```diff
@@ -42,7 +42,7 @@ protected void handleOnDestroy() {
     @SuppressWarnings("MissingPermission")
     @PluginMethod
     public void getStatus(PluginCall call) {
-        if (hasRequiredPermissions()) {
+        if (isPermissionDeclared("Network")) {
```
Comment (jcesarmobile) [inline network/android/src/main/java/com/capacitorjs/plugins/network/NetworkPlugin.java:45]: there is no `Network` alias, that's the plugin name, I get a 
> isPermissionDeclared: No alias defined for Network or missing @CapacitorPlugin annotation.

**Review by jcesarmobile — CHANGES_REQUESTED** [review]
Fixed the conflicts so I could properly test.
Added a comment on Network plugin changes, the other two looks good.

File: network/android/src/main/java/com/capacitorjs/plugins/network/NetworkPlugin.java (hunk 1 above)
Comment (carlpoole) [reply network/android/src/main/java/com/capacitorjs/plugins/network/NetworkPlugin.java]: Thanks. I dont think this plugin needs a permission annotation - the permission isn't dangerous and doesn't require permission checks from the user. We put this permissions in the manifest of the plugin now so it should always be there.

Comment (jcesarmobile) [reply network/android/src/main/java/com/capacitorjs/plugins/network/NetworkPlugin.java]: yeah, that was what I commented on the first implementation PR, but @ikeith said it did no harm to keep the check

https://github.com/ionic-team/capacitor-plugins/pull/8/files#r477256795

**Review by dwieeb — APPROVED** [review]

**Review by jcesarmobile — APPROVED** [review]

**Pull request merged**